config/connector.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config // import "go.opentelemetry.io/collector/config"
+import (
+	"go.opentelemetry.io/collector/component"
+)
+
+// ConnectorSettings defines common settings for a component.Connector configuration.
+// Specific exporters can embed this struct and extend it with more fields if needed.
+//
+// It is highly recommended to "override" the Validate() function.
+//
+// When embedded in the exporter config, it must be with `mapstructure:",squash"` tag.
+type ConnectorSettings struct {
+	id component.ID `mapstructure:"-"`
+	component.ConnectorConfig
+}
+
+// NewConnectorSettings return a new ConnectorSettings with the given ComponentID.
+func NewConnectorSettings(id component.ID) ConnectorSettings {
+	return ConnectorSettings{id: id}
+}
+
+var _ component.ConnectorConfig = (*ConnectorSettings)(nil)
+
+// ID returns the receiver ComponentID.
+func (cs *ConnectorSettings) ID() component.ID {
+	return cs.id
+}
+
+// SetIDName sets the receiver name.
+func (cs *ConnectorSettings) SetIDName(idName string) {
+	cs.id = component.NewIDWithName(cs.id.Type(), idName)
+}
+
+// Validate validates the configuration and returns an error if invalid.
+func (cs *ConnectorSettings) Validate() error {
+	return nil
```

Comment:
Mark as deprecated, see others.

Reply:
I'll just not add it then, right? Or is it preferred to add it as deprecated for consistency?

Reply:
Fine to not have it.